application/src/main/kotlin/net/corda/v5/application/persistence/PagedQuery.kt

```diff
@@ -0,0 +1,24 @@
+package net.corda.v5.application.persistence
+
+/**
+ * Used to build a Query that supports limit and offset.
+ *
+ * @param R the type of the results.
+ */
+interface PagedQuery<R> : Query<R> {
```

Comment:
I assume the default offset is 0, but what about the default limit? If that is unlimited, should you also specify a way to explicitly set that using the method for consistency?

I think the defaults should be documented also.